web/packages/teleterm/src/ui/DocumentAuthorizeWebSession/DocumentAuthorizeWebSession.tsx

```diff
@@ -0,0 +1,188 @@
+    documentsService.close(props.doc.uri);
+    ctx.notificationsService.notifyInfo(
+      'Web session has been opened in the browser'
+    );
+  }
+
+  return (
+    <Document visible={props.visible}>
+      <Flex
+        flexDirection="column"
+        maxWidth="680px"
+        width="100%"
+        mx="auto"
+        mt="4"
+        px="5"
+      >
+        <H1 mb="4">Authorize Web Session</H1>
+        <Flex flexDirection="column" gap={3}>
+          {!canAuthorize && (
+            <Alert
+              mb={0}
+              details={
+                <>
+                  To authorize a web session, you must first{' '}
+                  <a
+                    href="https://goteleport.com/docs/admin-guides/access-controls/device-trust/guide/#step-22-enroll-device"
+                    target="_blank"
+                  >
+                    enroll your device.
+                  </a>{' '}
+                  Then log out of the app, log back in, and try again.
+                </>
+              }
+            >
+              This device is not trusted
+            </Alert>
+          )}
+          {attempt.status === 'error' && (
+            <Alert mb={0} details={attempt.statusText}>
+              Could not authorize the session
+            </Alert>
+          )}
+          <Text>
+            Would you like to authorize a device trust web session for{' '}
+            <b>{clusterName}</b>?
+            <br />
+            The session will automatically open in a new browser tab.
+          </Text>
+          <Flex flexDirection="column" gap={2}>
+            <ButtonPrimary
+              disabled={
+                !canAuthorize ||
+                attempt.status === 'processing' ||
+                attempt.status === 'success'
+              }
+              size="large"
+              onClick={authorizeAndCloseDocument}
+            >
+              {getButtonText(attempt)}
+            </ButtonPrimary>
+            <ButtonText
+              disabled={
+                attempt.status === 'processing' || attempt.status === 'success'
+              }
+              onClick={openUnauthorizedAndCloseDocument}
+            >
+              Open Session Without Device Trust
+            </ButtonText>
+          </Flex>
+        </Flex>
+      </Flex>
+    </Document>
+  );
+}
+
+const confirmPath = 'webapi/devices/webconfirm';
+
+function buildAuthorizedSessionUrl(
+  rootCluster: Cluster,
+  webSessionRequest: WebSessionRequest,
+  confirmationToken: DeviceConfirmationToken
+): string {
+  const { redirectUri } = webSessionRequest;
+  let url = `https://${rootCluster.proxyHost}/${confirmPath}?id=${confirmationToken.id}&token=${confirmationToken.token}`;
+  if (redirectUri) {
+    url = `${url}&redirect_uri=${redirectUri}`;
+  }
+  return url;
+}
+
+function buildUnauthorizedSessionUrl(
+  rootCluster: Cluster,
+  webSessionRequest: WebSessionRequest
+): string {
+  const processedRedirectUri = processRedirectUri(
```

Comment:
It's because in `buildUnauthorizedSessionUrl` we don't return `redirect_uri` but instead a full path that we open directly. 
This was originally done this way in https://github.com/gravitational/teleport/pull/47221, I didn't change it.